userland/src/add_two_ints_client.cpp

```diff
@@ -0,0 +1,43 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <sys/time.h>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <simple_msgs/AllBuiltinTypes.h>
+#include <simple_msgs/AllDynamicArrayTypes.h>
+#include <simple_msgs/AllPrimitiveTypes.h>
+#include <simple_msgs/AllStaticArrayTypes.h>
+#include <simple_msgs/Nested.h>
+#include <simple_msgs/String.h>
+#include <simple_msgs/Uint32.h>
+
+#include <userland_msgs/AddTwoInts.h>
+
+int main(int argc, char** argv)
+{
+  rclcpp::init(argc, argv);
+
+  auto node = rclcpp::Node::make_shared("add_two_ints_client");
+
+  auto client = node->create_client<userland_msgs::AddTwoInts>("add_two_ints");
```

Comment:
We talked about the issues of spinning inside of a client, you'll need a recursive lock and a way to deal with multithreaded execution...

You can spin in different threads, for example you could create thread to call spin in or you could create the async request in a callback for subscription or you could spin in the main thread and then do the async request in a user created thread. There are lots of options.

The callback pattern that I described above has the advantage that you don't have to poll the future, so its more like request it and forget it. The future pattern has the advantage that you can use one thread to do work while periodically checking the future. But in both cases you need to be spinning in the mean time. In your example you just happen to use the single thread for polling the future and spinning.